object/types.proto

```diff
@@ -25,25 +26,25 @@ message UserHeader {
 message Header {
     oneof Value {
         // Link to other objects
-        Link Link                         = 1;
+        Link Link                              = 1;
         // Redirect not used yet
-        refs.Address Redirect             = 2;
+        refs.Address Redirect                  = 2;
         // UserHeader is a set of KV headers defined by user
-        UserHeader UserHeader             = 3;
+        UserHeader UserHeader                  = 3;
         // Transform defines transform operation (e.g. payload split)
-        Transform Transform               = 4;
+        Transform Transform                    = 4;
         // Tombstone header that set up in deleted objects
-        Tombstone Tombstone               = 5;
+        Tombstone Tombstone                    = 5;
         // Verify header that contains session public key and user's signature
-        session.VerificationHeader Verify = 6;
+        session.VerificationHeader Verify      = 6;
         // HomoHash is a homomorphic hash of original object payload
-        bytes HomoHash                    = 7 [(gogoproto.customtype) = "Hash"];
+        bytes HomoHash                         = 7 [(gogoproto.customtype) = "Hash"];
         // PayloadChecksum of actual object's payload
-        bytes PayloadChecksum             = 8;
+        bytes PayloadChecksum                  = 8;
         // Integrity header with checksum of all above headers in the object
-        IntegrityHeader Integrity         = 9;
+        IntegrityHeader Integrity              = 9;
         // StorageGroup contains meta information for the data audit
-        StorageGroup StorageGroup         = 10;
+        storagegroup.StorageGroup StorageGroup = 10;
```

Comment:
If you don't like the naming, consider something like `storagegroup.Header StorageGroup = 10;`, maybe it will be better.

Reply:
Let it be `storagegroup.StorageGroup StorageGroup` for now. Let's postpone one of the two most hard-solving CS problems until beta-release refactoring.